packages/relay-compiler/codegen/RelayFileWriter.js

```diff
@@ -353,13 +360,39 @@ class RelayFileWriter implements FileWriterInterface {
       return allOutputDirectories;
     });
   }
-}
 
-function md5(x: string): string {
-  return crypto
-    .createHash('md5')
-    .update(x, 'utf8')
-    .digest('hex');
+  /**
+   * Find all *.queryMap.json and write it into a single file.
+   * @param allOutputDirectories
+   */
+  writeCompleteQueryMap(
+    allOutputDirectories: Map<string, CodegenDirectory>,
+  ): void {
+    const queryMapFilePath = `${this._config.baseDir}/queryMap.json`;
+    try {
+      let queryMapJson = {};
+      allOutputDirectories.forEach(d => {
+        fs.readdirSync(d._dir).forEach(f => {
+          if (f.endsWith('queryMap.json')) {
+            const singleQueryMap = JSON.parse(
+              fs.readFileSync(path.join(d._dir, f), 'utf8'),
+            );
+            queryMapJson = {
+              ...queryMapJson,
+              ...singleQueryMap,
+            };
+          }
+        });
+      });
+
+      fs.writeFileSync(queryMapFilePath, JSON.stringify(queryMapJson));
+      this._reporter.reportMessage(
+        `Written complete queryMap file to ${queryMapFilePath}`,
+      );
```

Comment:
I haven’t looked into how other files are written to disk by Relay, but is it normal to use synchronous access here instead of async?

Reply:
Other areas in Relay are using sync as well for example `RelayCompilerCache` line 55 and `CodegenDirectory` line 213. I think it's possible to change this to async, although I think it won't really make a significant impact. Generally I tend to use sync for file operations than async just to avoid file locking and concurrency issues. Imho.